components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/Utils.java

```diff
@@ -85,10 +88,12 @@ public static void sendFault(MessageContext messageContext, int status) {
                     messageContext.getProperty("error_message_type").toString().equalsIgnoreCase("application/json")) {
                 axis2MC.setProperty(Constants.Configuration.MESSAGE_TYPE, "application/json");
             }*/
+
+            /* commented due to, fault message type need to be request accept type */
             //adding this fix to support any message type as error message type
-            if (messageContext.getProperty("error_message_type") != null) {
+            /*if (messageContext.getProperty("error_message_type") != null) {
```

Comment:
commented because it set the fault response as xml always. Since it is already fixed, remove the comment in commit: https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea